concordium-consensus/testdata/contracts/v1/checkpointing.wat

```diff
@@ -190,8 +190,9 @@
         (i32.const 1))
     ;; delete [000]
     (call $state_delete_entry (i32.const 0) (i32.const 3))
-    ;; delete the iter
-    (call $state_iterator_delete (i64.const 0)) ;; the iter created in the first receive function.
+    ;; attempt to delete the iterator created by the initial call to a_modify_proxy
+    ;; this should fail since iterators are per entrypoint
```

Comment:
Per entrypoint or per call to entrypoint?
```rust
enum IteratorAction { Create, Delete };

fn foo(action: IteratorAction) {
    match action {
        IteratorAction::Create => {create_iterator(0); foo(IteratorAction::Delete);}
        IteratorAction::Delete => { delete_iterator(0); } // Will this be valid or not?
}

foo(IteratorAction::Create);
```

Reply:
Well, per entry to the entrypoint :P

Reply:
I did not consider the other interpretation of what I wrote :)